File: sellergoods-service/src/main/java/com/hapiio/sellergoodsservice/controller/BrandController.java
package com.hapiio.sellergoodsservice.controller;

import com.hapiio.commons.common.PageResult;
import com.hapiio.commons.common.Result;
import com.hapiio.pojo.model.Brand;
import com.hapiio.sellergoodsservice.service.BrandService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Autowired
    private BrandService brandService;

    //查询品牌列表
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }

    //查询所有
    @RequestMapping("/findAll")
    public List<Brand> findAll(){
        return brandService.finAll();
    }

    //分页查询
    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNo,Integer pageSize){
        return brandService.findPage(pageNo,pageSize);
    }

    //添加
    @PostMapping("/add")
    public Result addBrand(@RequestBody Brand brand){
        return brandService.addBrand(brand);
    }

    //修改
    @PostMapping("/update")
    public Result updateBrand(@RequestBody Brand brand) {
        return brandService.updateBrand(brand);
    }

    //查询
    @GetMapping("/findOne")
    public Brand findBrand(Long id){
        return brandService.findBrand(id);
    }

    //删除
    @RequestMapping("/delete")
    public Result deleteBrand(@RequestParam("ids") Long[] ids) {
        return brandService.deleteBrand(ids);
    }

    //模糊查询
    @PostMapping("/search")
    public PageResult searchBrand(Integer pageNo,Integer pageSize,@RequestBody Brand brand){
        System.out.println(brand.getName());
        return brandService.search(pageNo,pageSize,brand);
    }


}

File: manager-web/src/main/resources/static/js/Controller/brandController.js
app.controller("brandController",function ($scope,brandService,$controller){

    $controller('baseController',{$scope:$scope});

    $scope.entity={};
    $scope.searchEntity={};

    $scope.findAll = function () {
        brandService.findAll().success(function (data) {
            $scope.list=data;
        })
    };

    $scope.findPage = function (pageNo,pageSize) {
        brandService.findPage(pageNo,pageSize).success(function (data) {
            $scope.list = data.rows;
            $scope.paginationConf.totalItems = data.total;

        })
    };

    var method = {};
    $scope.save = function(){
        if ($scope.entity.id == undefined||$scope.entity.id == ''|| $scope.entity.id == null){
            method = brandService.add;
        }else {
            method = brandService.update;
        }
        method($scope.entity).success(function (data) {
            if(data.success){
                console.log(data.message);
                //清空实体类
                $scope.entity = {};
                //重新加载页面
                $scope.reloadList();
            }else {
                console.log(data.message)
            }

        })
    };

    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (data) {

            if (data == undefined||data == ""||data == null){
                console.log("查找失败");
            }else {
                $scope.entity = data;
            }
        })
    };

    $scope.search = function (pageNo,pageSize) {
        brandService.search(pageNo,pageSize,$scope.searchEntity).success(function (data) {
            $scope.list = data.rows;
            $scope.paginationConf.totalItems = data.total;
        })
    };

    $scope.delete = function () {
       brandService.delete($scope.ids).success(function (data) {
            if (data.success){
                $scope.reloadList();
            }else {
                console.log(data.message);
            }
        })
    };

})